File: src/main/java/systems/conduit/main/core/plugin/annotation/PluginMeta.java
package systems.conduit.main.core.plugin.annotation;

import systems.conduit.main.core.plugin.config.Configuration;
import systems.conduit.main.core.plugin.config.NoConfig;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface PluginMeta {

    String name();
    String description();
    String version();
    String author();
    Dependency[] dependencies() default {};
    Class<? extends Configuration> config() default NoConfig.class;
    boolean reloadable() default false;

}

File: src/main/kotlin/systems/conduit/core/plugin/annotation/PluginMeta.kt
package systems.conduit.core.plugin.annotation

import systems.conduit.core.plugin.config.Configuration
import systems.conduit.core.plugin.config.NoConfig
import kotlin.reflect.KClass

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class PluginMeta(val name: String, val description: String = "", val version: String, val dependencies: Array<Dependency> = [],
                            val config: KClass<out Configuration> = NoConfig::class, val reloadable: Boolean = false)